Shared/Objects/DeviceProfileBuilder.swift

```diff
@@ -44,105 +17,166 @@ class DeviceProfileBuilder {
-                )]
-            } else {
-                transcodingProfiles = [TranscodingProfile(
-                    audioCodec: "aac,mp3,wav,eac3,ac3,opus",
-                    isBreakOnNonKeyFrames: true,
-                    container: segmentContainer,
-                    context: .streaming,
-                    maxAudioChannels: "6",
-                    minSegments: 2,
-                    protocol: "hls",
-                    type: .video,
-                    videoCodec: "h264,mpeg4"
-                )]
-            }
-        }
+        var codecProfiles: [CodecProfile] = []
+        var subtitleProfiles: [SubtitleProfile] = []
 
-        // Device supports FLAC?
-        if supportsFeature(minimumSupported: .A10X) {
+        // Separate native player profile from VLCKit profile
+        if Defaults[.VideoPlayer.videoPlayerType] == .swiftfin { // VLCKit
+            // Build direct play profiles
+            directPlayProfiles = [
+                // Just make one profile because if VLCKit can't decode it in a certain container, ffmpeg probably can't decode it for
+                // transcode either
+                DirectPlayProfile(
+                    // No need to list containers or videocodecs since if jellyfin server can detect it/ffmpeg can decode it, so can VLCKit
+                    // However, list audiocodecs because ffmpeg can decode TrueHD/mlp but VLCKit cannot
+                    audioCodec: "flac,alac,aac,eac3,ac3,dts,opus,vorbis,mp3,mp2,mp1,pcm_s24be,pcm_s24le,pcm_s16be,pcm_s16le,pcm_u8,pcm_alaw,pcm_mulaw,pcm_bluray,pcm_dvd,wavpack,wmav2,wmav1,wmapro,wmalossless,nellymoser,speex,amr_nb,amr_wb",
+                    type: .video
+                ),
+            ]
+
+            // Build transcoding profiles
+            // The only cases where transcoding should occur:
+            // 1) TrueHD/mlp audio
+            // 2) When server forces transcode for bitrate reasons
             transcodingProfiles = [TranscodingProfile(
-                audioCodec: "aac,mp3,wav,ac3,eac3,flac,opus",
+                audioCodec: "flac,alac,aac,eac3,ac3,dts,opus,vorbis,mp3,mp2,mp1",
+                // no PCM,wavpack,wmav2,wmav1,wmapro,wmalossless,nellymoser,speex,amr_nb,amr_wb in mp4
                 isBreakOnNonKeyFrames: true,
-                container: segmentContainer,
+                container: "mp4",
                 context: .streaming,
-                maxAudioChannels: "6",
+                maxAudioChannels: "8",
                 minSegments: 2,
                 protocol: "hls",
                 type: .video,
-                videoCodec: "hevc,h264,mpeg4"
+                videoCodec: "hevc,h264,av1,vp9,vc1,mpeg4,h263,mpeg2video,mpeg1video,mjpeg" // vp8,msmpeg4v3,msmpeg4v2,msmpeg4v1,theora,ffv1,flv1,wmv3,wmv2,wmv1
+                // not supported in mp4
             )]
-        }
 
-        var codecProfiles: [CodecProfile] = []
+            // Create subtitle profiles
+            subtitleProfiles = [
```

Comment:
Still an issue with subs - pgssub is throwing server error; not sure if something has changed server-side in 10.8.10. Regardless, subs are messy because I don't think we want to have to strip out embedded subs just to play the movie - should probably only do that on demand when sub is requested.